lib/src/bottom_sheet.dart

```diff
@@ -240,22 +238,27 @@ class RubberBottomSheetState extends State<RubberBottomSheet> with TickerProvide
         else if (controller.value < controller.lowerBound) {
           diff = controller.lowerBound - controller.value;
         }
-        if(controller.value < controller.upperBound && controller.dismissable && controller.animationState.value == AnimationState.expanded) {
+        if (controller.value < controller.upperBound &&
+            controller.dismissable &&
+            controller.animationState.value == AnimationState.expanded) {
           diff = controller.upperBound - controller.value;
         }
         if (diff != null) {
           friction = widget.dragFriction * pow(1 - diff, 2);
         }
 
-        controller.value -= details.primaryDelta / _screenHeight * friction;
-        if(_shouldScroll && controller.value >= controller.upperBound && !_draggingPeak(_lastPosition)) {
+        controller.value -= details.primaryDelta / (_screenHeight * friction);
```

Comment:
The animation naturally exceeds the limits because it's a spring, right? So it bounces passed the target. The original would does so as well, with `friction = 1`.

This is actually why I added the `padding`, so that the animation could overshoot the target based on the spring, but bounce off screen. It still results a "pausing" effect at the min and max though. I think using the `Animation` api instead of the `Simulation` api would be a lot easier - maybe leveraging [sprung](https://pub.dev/packages/sprung) or something